src/client/auth/actions.js

```diff
@@ -33,14 +33,24 @@ function validateForm(fields) {
 }
 
 function validateCredentials(fields) {
-  // Simulate lengthy async action.
   return new Promise((resolve, reject) => {
-    setTimeout(() => {
-      if (fields.password === 'pass1')
+
+    var xhr = new XMLHttpRequest();
+
+    xhr.open('POST', '/api/v1/auth/login', true);
+    xhr.setRequestHeader('Content-type', 'application/json');
+
+    // @TODO show how to handle different password/username server errors
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState !== 4) return;
+      if (xhr.status === 200) {
         resolve(fields);
```

Comment:
The tests are failing because of `unnecessary` curly brace after `if` condition which is really stupid in my opinion. I prefer always explicitly leaving them to avoid regression errors but can change that so the tests pass. Otherwise, I'd suggest disabling that rule in `eslint`. Apart from that all good.